File: FantasyRadio/src/main/java/ru/sigil/fantasyradio/utils/TrackFactory.java
package ru.sigil.fantasyradio.utils;

import ru.sigil.fantasyradio.BackgroundService.ITrack;
import ru.sigil.fantasyradio.BackgroundService.ITrackFactory;
import ru.sigil.fantasyradio.saved.MP3Entity;

/**
 * Created by namelessone
 * on 04.01.17.
 */

public class TrackFactory implements ITrackFactory{
    @Override
    public ITrack createTrack(String author, String title, String recDirectory, String time) {
        return new MP3Entity(author, title, recDirectory, time);
    }
}

File: FantasyRadio/src/main/kotlin/ru/sigil/fantasyradio/utils/TrackFactory.kt
package ru.sigil.fantasyradio.utils

import ru.sigil.fantasyradio.saved.MP3Entity
import ru.sigil.bassplayerlib.ITrack
import ru.sigil.bassplayerlib.ITrackFactory

/**
 * Created by namelessone
 * on 07.12.18.
 */
class TrackFactory : ITrackFactory {
    override fun createTrack(author: String?, title: String?, recDirectory: String, time: String?): ITrack = MP3Entity(author, title, recDirectory, time)
}
